tests/drivers/uart/uart_async_dual/testcase.yaml

```diff
@@ -0,0 +1,13 @@
+common:
+  tags:
+    - drivers
+    - uart
+tests:
+  drivers.uart.async_dual:
+    harness: ztest
+    harness_config:
+      fixture: gpio_loopback
```

Comment:
Might be a good idea to add another `fixture` as requirements since the test needs two UARTS connecting to each other.